### PR TITLE
lib/model: Don't mark deleted local only as locally changed (fixes #6432)

### LINKED ISSUE #6432: Deleted file that existed locally only reported as locally changed
A file that only ever existed locally, but is now deleted, keeps showing up as a locally changed item. That has no value, as for all practical intents and purposes it does not exist. See https://forum.syncthing.net/t/how-to-accept-local-changes/14657/47?u=imsodin and posts leading up to it (thanks @xarx00).

### PR DESCRIPTION
**Pull request by imsodin**
> A file that only ever existed locally, but is now deleted, keeps showing up as a locally changed item. That has no value, as for all practical intents and purposes it does not exist.

Like with similar special conditions on send only folder, this adds a stage to pulling that checks for this situation and fixes it by removing the local flag.

### CODE REVIEW

File: lib/db/structs.go [hunk 1]
```diff
@@ -129,19 +129,19 @@ func (f FileInfoTruncated) FileModifiedBy() protocol.ShortID {
...
 func (f FileInfoTruncated) ConvertToDeletedFileInfo(by protocol.ShortID) protocol.FileInfo {
-	file := f.copyToFileInfo()
+	file := f.CopyToFileInfo()
 	file.SetDeleted(by)
 	return file
 }
 
 // copyToFileInfo just copies all members of FileInfoTruncated to protocol.FileInfo
-func (f FileInfoTruncated) copyToFileInfo() protocol.FileInfo {
+func (f FileInfoTruncated) CopyToFileInfo() protocol.FileInfo {
```
Comment (calmh) [inline lib/db/structs.go]: Somewhat skeptical of exposing this method given the latest change to putFile and the fact that the resulting FileInfo can't be easily or safely passed back into the database layer. The Deleted/Ignored cases above are the exceptions where this is safe (as those shouldn't have block lists).

Comment (imsodin) [reply lib/db/structs.go]: I added a new method that converts deleted, truncated file infos to protocol file infos and panics if the file isn't actually deleted.

**Comment by imsodin** [comment]
lgtm?

**Comment by calmh** [comment]
A test for this situation would be cool...

**Comment by imsodin** [comment]
I wrote the test and it passes on master - in hindsight obviously: Deleted files that only we had are handled by scanning (already tested),  and if the remote deletes a file we already deleted we pull that deletion thus removing the locally changed item. I'll file the test separately.

**Comment by calmh** [comment]
So test revealed there was no bug, or bug is something other than we though?

**Comment by imsodin** [comment]
Initially I thought it was "no bug", but right now I am writing another test as I have suspicion for "https://github.com/syncthing/syncthing/pull/6433#issuecomment-604987278" :)

**Pull request closed without merging**